backend/infrastructure/api/endpoints/product.py

```diff
@@ -0,0 +1,69 @@
+from fastapi import APIRouter, UploadFile, File, HTTPException, Form
+from infrastructure.mongo.product_handler import upload_product, get_product, get_image
+from fastapi.responses import StreamingResponse
+from io import BytesIO
+
+router = APIRouter()
+
+@router.post("/upload")
+async def upload_product_endpoint(
+    name: str = Form(...),
+    price: float = Form(...),
+    country_of_origin: str = Form(...),
+    description: str = Form(...),
+    fruit_or_vegetable: str = Form(...),
+    file: UploadFile = File(...),
+    expiry_date: str = Form(...),
+):
+    try:
```

Comment:
this whole try should be in validation/service component, which extensions of files do we support? please provide validation for this